plugins/org.eclipse.n4js.cli/src/org/eclipse/n4js/cli/N4jscOptions.java

```diff
@@ -137,11 +137,17 @@
 		int maxWarns = 0;
 
 		@Option(name = "--clean", aliases = "-c", //
-				usage = "[compile] output folders are cleaned before compilation.", //
+				usage = "[compile|lsp] output folders are cleaned at start.", //
 				handler = N4JSCmdLineParser.N4JSBooleanOptionHandler.class)
-		@GoalRequirements(goals = N4jscGoal.compile)
+		@GoalRequirements(goals = { N4jscGoal.compile, N4jscGoal.lsp })
 		boolean clean = false;
 
+		@Option(name = "--noPersist", aliases = "-np", //
+				usage = "[compile|lsp] disable persisting of type index to disk.", //
+				handler = N4JSCmdLineParser.N4JSBooleanOptionHandler.class)
```

Comment:
Does this mean persisting the type index is on by default even for goal "compile"? I think that would be counter-intuitive.

Reply:
For the record: we just talked about that and chose to leave it like that for the moment.